frontend/src/lib/apolloClient.ts

```diff
@@ -0,0 +1,94 @@
+import { ApolloClient, HttpLink, InMemoryCache, NormalizedCacheObject } from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
+import { config } from "@utils/config";
+import cookie from "cookie";
+import merge from "deepmerge";
+import { IncomingMessage } from "http";
+import isEqual from "lodash/isEqual";
+import { GetServerSidePropsContext } from "next";
+import { useMemo } from "react";
+
+type PageProps = {
+  props: any;
```

Comment:
This is obviously not ideal by any means, however, I have not been able to find a solution to this issue. If this is typed as e.g.
```ts
type PageProps = {
  props: Record<string, any>;
}
```
Then `getServerSideProps` will complain that it does not match the expected type, but I haven't been able to figure out exactly how to deal with this.

Reply:
It looks to me that by examining when this is actually used, the following information can be deduced:

The props input to `App` in `_app.tsx` are declared as
```type AppPropsWithError = AppProps & { err: Error };```
Where `AppProps` are declared as 
```
AppInitialProps & {
    Component: NextComponentType<NextPageContext, any, P>;
    router: R;
    __N_SSG?: boolean;
    __N_SSP?: boolean;
    __N_RSC?: boolean;
}; 
```

and `AppInitialProps` is declared as 
```
export declare type AppInitialProps = {
    pageProps: any;
};
```

`AppPropsWithError` is destructured as follows
```const App = ({ Component, pageProps, err }: AppPropsWithError): JSX.Element => {```

So `pageProps` is simply `any` as declared by next.js. One option is then to type it as `AppInitialProps`however, then the property `props` does not exists, which makes me question whether my deduction here is incorrect 🤔